### PR TITLE
respect shareapi_allow_share_dialog_user_enumeration in Principal backend for Sabre/DAV

### LINKED ISSUE #11087: autocompletion leaks data in different components even if autocompletion is deactivated
Several components and third party apps do no honor the setting in sharing "Allow username autocompletion in share dialog. If this is disabled the full username or email address needs to be entered." 

While this works for the file sharing dialog it doesn't for several (if not the majority) of other applications. This means that running a public nextcloud with more then the core apps i impossible.

While I could file a bug report for every app where I discover this problem, I think the nexcloud core should be responsible for not leaking data. There are several similar problems reported in the past for different applications and circumstances: #6530,  #5107, #8392, #7428

Not reported is this so far for e.g. calendar app or polls (as long as I can see).

### Expected behaviour
If autocompletion feature is deactivated users should not see each other in any sharing dialog as long as they don't know the complete username / email-adress, regardless of the authentication backend (database, ldap, ...). This feature should be honored in every app.

### Actual behaviour
Some apps (file sharing, contacts) honor the settings, some not.

### Server configuration

**Nextcloud version:** (see Nextcloud admin page)

Nextcloud 13.0.6 and earlier versions

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #11087

### CODE REVIEW

File: apps/dav/lib/RootCollection.php [hunk 1]
```diff
@@ -63,9 +63,10 @@ public function __construct() {
 			$shareManager,
 			\OC::$server->getUserSession(),
 			\OC::$server->getAppManager(),
-			$proxyMapper
+			$proxyMapper,
+			\OC::$server->getConfig()
 		);
-		$groupPrincipalBackend = new GroupPrincipalBackend($groupManager, $userSession, $shareManager, $l10n);
+		$groupPrincipalBackend = new GroupPrincipalBackend($groupManager, $userSession, $shareManager);
```
Comment (georgehrke) [inline apps/dav/lib/RootCollection.php]: GroupPrincipalBackend doesn't even take a fourth parameter:
https://github.com/nextcloud/server/blob/master/apps/dav/lib/DAV/GroupPrincipalBackend.php#L55

Not sure why it was there in the first place.

**Review by rullzer — APPROVED** [review]
:elephant:

**Review by tcitworld — APPROVED** [review]

**Comment by rullzer** [comment]
Rebase ;)

**Comment by georgehrke** [comment]
Rebased

**Comment by georgehrke** [comment]
GRMPF

**Comment by georgehrke** [comment]
Backport to stable17: https://github.com/nextcloud/server/pull/18205

**Comment by juliushaertl** [comment]
@georgehrke Do we still need a backport for 16 as well?

**Comment by georgehrke** [comment]
ideally yes, let me look into it, so we can include it in the maintenance releases.

**Pull request merged**